libpod/options.go

```diff
@@ -2342,3 +2440,82 @@ func WithVolatile() CtrCreateOption {
 		return nil
 	}
 }
+
+// WithCpusCtr computes the CPU period and quota given the CPU core count for a pod
+func WithCPUs(num float64) PodCreateOption {
+	return func(pod *Pod) error {
+		if pod.valid {
+			return define.ErrPodFinalized
+		}
+		if num > float64(runtime.NumCPU()) {
+			return errors.New("Invalid CPU amount, exceeds total number available")
+		}
+
+		pod.config.CPUs = num
+		return nil
+	}
+}
+
+// WithCpuSetCpusCtr computes and sets the Cpus linux resource string which determines the amount of cores, from those available,  we are allowed to execute on
+func WithCPUSetCPUs(inp string) PodCreateOption {
+	return func(pod *Pod) error {
+		if pod.valid {
+			return define.ErrPodFinalized
+		}
+		ret, err := parsers.ParseUintList(inp)
+		if err != nil {
+			errors.Wrapf(err, "could not parse list")
+		}
+		for i := 0; i < len(ret); i++ {
+			if ret[i] {
+				if i > runtime.NumCPU() {
```

Comment:
```suggestion
				if i > sysinfo.NumCPU() {
```
I believe this is more compatible, only gotcha is you need to fail back to runtime.NumCPU() when sysinfo.NumCPU() returns 0